Source/WebKit/NetworkProcess/NetworkSession.cpp

```diff
@@ -927,4 +930,24 @@ Ref<NetworkBroadcastChannelRegistry> NetworkSession::protectedBroadcastChannelRe
     return m_broadcastChannelRegistry;
 }
 
+#if ENABLE(CONTENT_EXTENSIONS)
+WebCore::ResourceMonitorThrottler& NetworkSession::resourceMonitorThrottler()
+{
+    if (!m_resourceMonitorThrottler)
+        m_resourceMonitorThrottler = WebCore::ResourceMonitorThrottler::create();
+
+    return *m_resourceMonitorThrottler;
+}
+
+Ref<WebCore::ResourceMonitorThrottler> NetworkSession::protectedResourceMonitorThrottler()
+{
+    return resourceMonitorThrottler();
+}
+
+void NetworkSession::resetResourceMonitorThrottlerForTesting()
+{
+    m_resourceMonitorThrottler = nullptr;
```

Comment:
It might make more sense to call a `reset` method on the resource monitor throttler here rather than destroying and recreating the object.

Presumably if this is eventually backed by a DB, you'd want to pass in a completion handler in to this method which only returns when the DB has finished deleting data on disk for instance.